reposerver/repository/repository_test.go

```diff
@@ -759,17 +764,25 @@ func TestGenerateHelmWithValuesDirectoryTraversal(t *testing.T) {
 	assert.NoError(t, err)
 
 	// Test the case where the path is "."
-	service = newService("./testdata/my-chart")
+	service = newService("./testdata")
 	_, err = service.GenerateManifest(context.Background(), &apiclient.ManifestRequest{
 		Repo:    &argoappv1.Repository{},
 		AppName: "test",
 		ApplicationSource: &argoappv1.ApplicationSource{
-			Path: ".",
+			Path: "./my-chart",
 		},
 	})
 	assert.NoError(t, err)
 }
 
+func TestChartRepoWithOutOfBoundsSymlink(t *testing.T) {
+	service := newService(".")
+	source := &argoappv1.ApplicationSource{Chart: "out-of-bounds-chart", TargetRevision: ">= 1.0.0"}
+	request := &apiclient.ManifestRequest{Repo: &argoappv1.Repository{}, ApplicationSource: source, NoCache: true}
+	_, err := service.GenerateManifest(context.Background(), request)
+	assert.ErrorContains(t, err, "chart contains out-of-bounds symlinks")
```

Comment:
Is it possible to use `assert.ErrorIs` here?

Reply:
Not that I'm aware of. At this point it's a new error and not my custom type.

Reply:
New as in it's wrapped? If you use `fmt.Errorf("... %w", wrapperErr)`, then `errors.As` will unwrap until it finds the error.

But don't sweat it.  :-)

Reply:
Oooooh. Learned me some more Go today!